src/dialogs/symlink.tsx

```diff
@@ -0,0 +1,244 @@
+            });
+
+            if (owner !== null) {
+                await cockpit.spawn(["chown", "--no-dereference", owner, symlinkName], {
+                    directory: path,
+                    superuser: "require"
+                });
+            }
+        } catch (exc) {
+            const err = exc as BasicError;
+            setErrorMessage(err.toString());
+            return;
+        }
+
+        dialogResult.resolve();
+    };
+
+    return (
+        <Modal
+          position="top"
+          title={fmt_to_fragments(_("Create link to $0"), <b>{selected.name}</b>)}
+          variant={ModalVariant.medium}
+          isOpen
+          className="file-symlink-modal"
+          onClose={() => dialogResult.resolve()}
+          footer={
+              <>
+                  <Button
+                    variant="primary"
+                    onClick={() => {
+                        const filename_invalid = checkLinkName(symlinkName, cwdInfo?.entries || {});
+                        if (filename_invalid) {
+                            setSymlinkError(filename_invalid);
+                        } else {
+                            createLink();
+                        }
+                    }}
+                    isDisabled={errorMessage !== undefined ||
+                        symlinkError !== null ||
+                        cwdInfo === null}
+                  >
+                      {_("Create link")}
+                  </Button>
+                  <Button variant="link" onClick={() => dialogResult.resolve()}>{_("Cancel")}</Button>
+              </>
+          }
+        >
+            {errorMessage &&
+            <InlineNotification
+              type="danger"
+              text={errorMessage}
+              isInline
+            />}
+            <Form
+              isHorizontal
+              onSubmit={e => {
+                  e.preventDefault();
+                  createLink();
+                  return false;
+              }}
+            >
+                <FormGroup fieldId="target-name" label={_("Target")}>
+                    <TextInput
+                      value={selected.name}
+                      readOnlyVariant="plain"
+                      id="target-name"
+                    />
+                </FormGroup>
+                <FormGroup fieldId="symlink-input" label={_("Symlink name")}>
+                    <TextInput
+                      validated={symlinkError ? "error" : "default"}
+                      value={symlinkName}
+                      onChange={(_, val) => {
+                          const filename_invalid = checkLinkName(val, cwdInfo?.entries || {});
+                          setSymlinkError(filename_invalid);
+                          setErrorMessage(undefined);
+                          setSymlinkName(val);
+                      }}
+                      id="symlink-input" autoFocus // eslint-disable-line jsx-a11y/no-autofocus
```

Comment:
This doesn't automatically select the "link to filename" text, so it has to be manually deleted now. Can't we do that as [`placeholder=`](https://v5-archive.patternfly.org/components/forms/text-input) instead? (This is a design question, and I don't claim authority here -- but we do use this a lot in cockpit and c-files).

Reply:
A placeholder is not auto selected right? I think I will make a follow up issue for this.

Reply:
Yes, the `autoFocus` should stay of course, it was just a convenient place to attach the comment to. You can't "select" the placeholder itself, it's not an actionable text.

Reply:
We can't do a place holder as the whole idea was that the default symlink target would be `Link to $0`. I will create a follow up issue as the idea is, to have it pre-select "link to" and allow you to quickly remove that once.

Reply:
But this isn't a sensible default -- nobody actually wants a `link to myfile`. It's an explanation, not a reasonable default. And there is no reasonable default, hence it should be empty.

Reply:
Compare this to "create new file" where we also don't pre-fill the file name.